horace_core/symop/@Symop/Symop.m

```diff
@@ -0,0 +1,369 @@
+        % Input:
+        %   obj    Array of symmetry operator objects
+        %   vec    3xN list of 3-vectors to transform
+        % Output:
+        %   vec    Transformed set of vectors
+
+            if size(vec, 1) ~= 3
+                error('HORACE:symop:invalid_argument', ...
+                      'Input must be list of 3-vectors')
+            end
+
+            for i = numel(obj):-1:1
+                vec = vec - obj(i).offset;
+                vec = obj(i).R * vec;
+                vec = vec + obj(i).offset;
+            end
+        end
+
+        function disp(obj)
+        % Display set of symmetry operations resulting in transform
+        % even if specified as array of symops
+            if isscalar(obj)
+                obj.local_disp();
+            else
+                disp('[');
+                for i = obj
+                    i.local_disp();
+                end
+                disp(']');
+            end
+        end
+
+        function pix = transform_pix(obj, pix)
+        % Transform pixel coordinates into symmetry related coordinates
+        %
+        % The transformation converts the components of a vector which is
+        % related by the symmetry operation into the equivalent vector. For example,
+        % if the symmetry operation is a rotation by 90 degrees about
+        % [0,0,1] in a cubic lattice with lattice parameter 2*pi, the point [0.3;0.1;2]
+        % is transformed into [0.1;-0.3;2].
+        %
+        %   >> pix = transform_pix (obj, pix_in)
+        %
+        % Input:
+        % ------
+        %   obj         Symmetry operator or array of symmetry operators
+        %               If an array, then they are applied in order obj(1), obj(2),...
+        %
+        %   upix_to_rlu Matrix to convert components of a vector in pixel coordinate
+        %              frame (which is an orthonormal frame) into rlu (3x3 matrix)
+        %
+        %   upix_offset Offset of origin of pixel coordinate frame (rlu) (vector length 3)
+        %
+        %   pix         PixelData object
+        %
+        % Output:
+        % -------
+        %   pix         Transformed PixelData object
+
+            % Check input
+            if ~isa(pix, 'PixelDataBase')
+                error('HORACE:Symop:invalid_argument', ...
+                      'transform_pix requires pixels');
+            end
+
+            % Get transformation
+            if isa(pix, 'PixelDataMemory')
+                sel = obj.in_irreducible(pix.q_coordinates);
+                pix.q_coordinates(:, ~sel) = obj.transform_vec(pix.q_coordinates(:, ~sel));
+            else
+                error('HORACE:Symop:not_implemented', ...
+                      'filebacked pix symmetry reduction not possible')
+            end
+
+        end
+
+        function [proj, pbin] = transform_proj (obj, proj, pbin)
+        % Transform projection axes description by the symmetry operation
+        %
+        %   >> proj = transform_proj (obj, proj)
```

Comment:
missing pbin

having said that, pbin is not used. Is this a superclass thing? needs flagging